src/main/kotlin/io/snyk/plugin/net/TokenInterceptor.kt

```diff
@@ -43,10 +44,24 @@ class TokenInterceptor(private var projectManager: ProjectManager? = null) : Int
         }
         request.addHeader("Accept", "application/json")
             .addHeader("Content-Type", "application/json")
-            .addHeader("User-Agent", "snyk-intellij-plugin/${pluginInfo.integrationVersion}")
+            .addHeader("User-Agent", getUserAgentString())
             .addHeader("x-snyk-ide", "${pluginInfo.integrationName}-${pluginInfo.integrationVersion}")
         return chain.proceed(request.build())
     }
+
+    fun getUserAgentString(): String {
+//      $APPLICATION/$APPLICATION_VERSION ($GOOS;$GOARCH[;$BINARY_NAME]) [$SNYK_INTEGRATION_NAME/$SNYK_INTEGRATION_VERSION [($SNYK_INTEGRATION_ENVIRONMENT/$SNYK_INTEGRATION_ENVIRONMENT_VERSION)]]
+        val integrationName = pluginInfo.integrationName
+        val integrationVersion = pluginInfo.integrationVersion
+        val integrationEnvironment = pluginInfo.integrationEnvironment
+        val integrationEnvironmentVersion = pluginInfo.integrationEnvironmentVersion
+        val os = SystemUtils.OS_NAME
```

Comment:
Question: should we try to map this to a common set of OS and Arch representations?

Reply:
We should once we have data from all different system and decide then to normalize it.